log/logtest/assert.go

```diff
@@ -75,3 +86,13 @@ func Transform[A, B any](f func(A) B) AssertOption {
 		return cfg
 	})
 }
+
+// Message sets a prepended assertion failure message.
+// The message is formatted with the arguments using fmt.Sprintf.
+func Message(msg string, args ...any) AssertOption {
```

Comment:
`logtest.Message` reads like it is testing the log message. Can we name this so it communicates that it is a failure message?

Reply:
Proposals:
- `FailMessage`
- `FailureMessage`
- `FailMsg`

What is your preference?

Reply:
What about `Description`? Meaning you pass `logtest.Description` as an option to describe the thing being tested. That way the error message reads:

```
<description of test>
mismatch (-want +got):
...
```

Calling it "failure message" something indicates to the user that they need to put the full failure reporting context. Thereby recreating the "mismatch (-want +got):" output.

Reply:
nice idea

Reply:
PTAL https://github.com/open-telemetry/opentelemetry-go/pull/6796/commits/d17443a6f53c276028b304261147dc4533ce642e